web/src/main/java/org/cbioportal/web/config/WebServletContextListener.java

```diff
@@ -0,0 +1,113 @@
+package org.cbioportal.web.config;
+
+import org.apache.commons.lang.StringUtils;
+import org.cbioportal.proxy.ProxyController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Value;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import java.io.*;
+import java.util.Properties;
+
+/**
+ * Created by Hongxin Zhang on 11/15/19.
+ */
+public class WebServletContextListener implements ServletContextListener, InitializingBean {
+    private Logger LOG = LoggerFactory.getLogger(WebServletContextListener.class);
+
+    private Boolean showOncokb;
+    
+    private String oncokbToken;
+    
+    private String oncokbPublicApiUrl;
+
+    @Override
+    public void contextDestroyed(ServletContextEvent arg0) {
+        System.out.println("ServletContextListener destroyed");
+    }
+
+    //Run this before web application is started
+    @Override
+    public void contextInitialized(ServletContextEvent arg0) {
+        Properties properties = loadProperties(getResourceStream("portal.properties"));
```

Comment:
@zhx828 i think this should be done in another way, because it doesn't support setting these properties on the command line with `-D`

Reply:
can use something like this from `GlobalProperties.java`: https://github.com/cBioPortal/cbioportal/blob/d786e77a59a880fc331e608276586a0b5f0b4c78/core/src/main/java/org/mskcc/cbio/portal/util/GlobalProperties.java#L350-L353

Reply:
yeah... I thought I put it in, will update.